### ISSUE TITLE
Allow disabling Bluetooth but keep GadgetBridge alive

### ISSUE THREAD

**Issue by Avamander** [Freeyourgadget/Gadgetbridge#595]
There are quite a few cases where it would be nice if GB just kept itself alive and just waited for BT to be enabled without going all crazy (closing itself or flickering) or toasting. 

For example the issue occurs when:

- Enabling Airplane mode and then reenabling BT for sleep tracking 
- Restarting Bluetooth if it behaves a bit wrong
- Disabling Bluetooth for a while to save battery

**Comment by cpfeiffer** [comment]
Need some more infos here:
First, what's the problem with Gadgetbridge shutting (mostly) down? Yes, it displays a toast, but "going crazy"?
Second, you don't need to be connected to your band/pebble, etc. to let it do sleep tracking.
Third, Gadgetbridge can automatically restart when you reenable Bluetooth.

**Comment by Avamander** [comment]
1) It's disruptive, inefficient, neither is it good UX. The UI shouldn't close like that when disabling Bluetooth, it should open a dialog box if anything.

The toasting is not necessary, "Bluetooth is disabled", well duh, I just disabled it and I see from my notifications that my gadget is not connected.

2) The sleep tracking is not Pebble's.

3) Indeed. That's okay, but I am speaking of while the app itself is open.

**Comment by danielegobbetti** [comment]
The issue message is basically a rant (read: there is no proposal of what to change with some reasoning but rather a list of things you find annoying).

The last comment does indeed contain some suggestions of how and why things could be changed, but one has to read through harsh words to get to them, and I know few people that in their free time like distilling good suggestions from sentences that evoke bad feelings on top of writing code.

Tl;Dr: Don't you think that using more kind words would yield to better acceptance of your proposals?

**Comment by Avamander** [comment]
I try to be as neutral as possible, clearly stating the issue, though that's what the issue box is for. The comments should contain the solutions. In addition to that, I can't have a solution for everything, people with better insight probably can suggest a better solution that I can.

**Comment by cpfeiffer** [comment]
1) Arguably we can spare the toast.
2) What does this have to do with Gadgetbridge then? Please elaborate.
3) Sorry, I don't get it. Gadgetbridge closes, so what do you mean with "while the app itself is open"?

If you disable BT, you typically can't do much with Gadgetbridge, except for looking at your data. So it is kinda sensible to close it. This will save memory and a little CPU as well (for system's memory management).

Since it's mostly the activities that are closed (all those broadcast listeners are managed by the OS) we don't save a lot of memory I guess, so maybe the memory savings are not a good reason anymore.

**Comment by ashimokawa** [comment]
The reason to close all activities was that most of them did not work without a connected device in the past. I guess we can spare to close the main activity now.

@Avamander 
Be nice

**Comment by Avamander** [comment]
@cpfeiffer 
3. You open the app, it shows you stuff, now disable Bluetooth, what happens?
2. That was just an example use case, it can be whatever watchapp with Android companion.

**Comment by ashimokawa** [comment]
GB no longer quits when disabling bluetooth in 0.19.2
